src/compat/db_admin.c

```diff
@@ -505,6 +506,22 @@ db_set_client_type (int client_type)
     }
 }
 
+char *
+db_get_client_ip_addr (void)
+{
+  return db_Client_ip_addr;
+}
+
+void
+db_set_client_ip_addr (const char *ip_addr)
+{
+  if (ip_addr)
+    {
+      memcpy (db_Client_ip_addr, ip_addr, 15);
+      db_Client_ip_addr[15] = '\0';
```

Comment:
Is this copying in case ip_addr is not a normal address (in case the length is 16 or more)?